### PR TITLE
U268/t269 compute animations

### PR DESCRIPTION
**Pull request by Piratkopia13**
now runs on gpu, ~100fps improvement in current scene

### CODE REVIEW

File: SPLASH/src/game/states/GameState.cpp [hunk 1]
```diff
@@ -872,25 +892,25 @@ void GameState::initAnimations() {
...
-		//auto animationEntity4 = ECS::Instance()->createEntity("animatedModel4");
-		//animationEntity4->addComponent<TransformComponent>();
-		//animationEntity4->getComponent<TransformComponent>()->translate(-1,2,-3);
-		//animationEntity4->getComponent<TransformComponent>()->translate(110.f, 100.f, 100.f);
-		//animationEntity4->getComponent<TransformComponent>()->scale(0.005f);
-		//animationEntity4->addComponent<ModelComponent>(&m_app->getResourceManager().getModel("AnimationTest/DEBUG_BALLBOT.fbx", shader));
-		//animationEntity4->getComponent<ModelComponent>()->getModel()->setIsAnimated(true);
-		//animationEntity4->addComponent<AnimationComponent>(&m_app->getResourceManager().getAnimationStack("AnimationTest/DEBUG_BALLBOT.fbx"));
-		//animationEntity4->getComponent<AnimationComponent>()->currentAnimation = animationEntity4->getComponent<AnimationComponent>()->getAnimationStack()->getAnimation(0);
+	//auto animationEntity4 = ECS::Instance()->createEntity("animatedModel4");
```
Comment (Skratzy) [inline SPLASH/src/game/states/GameState.cpp]: Remove unused code

File: Sail/src/API/DX12/DX12API.h [hunk 2]
```diff
@@ -143,8 +147,7 @@ class DX12API : public GraphicsAPI {
 
 	// Queues
 	wComPtr<ID3D12CommandQueue> m_directCommandQueue;
-	wComPtr<ID3D12CommandQueue> m_computeCommandQueue;
-	wComPtr<ID3D12CommandQueue> m_copyCommandQueue;
+	wComPtr<ID3D12CommandQueue> m_computeCommandQueueAnimations;
 
 	// Commands
 	//Command m_preCommand;
```
Comment (Skratzy) [inline Sail/src/API/DX12/DX12API.h]: Is this used?

File: Sail/src/API/DX12/shader/DX12ShaderPipeline.cpp [hunk 3]
```diff
@@ -150,7 +156,7 @@ void* DX12ShaderPipeline::compileShader(const std::string& source, const std::st
 		}
 		if (pShaders)
```
Comment (Skratzy) [inline Sail/src/API/DX12/shader/DX12ShaderPipeline.cpp]: Missing curly brackets

File: Sail/src/API/DX12/shader/DX12ShaderPipeline.cpp [hunk 4]
```diff
@@ -150,7 +156,7 @@ void* DX12ShaderPipeline::compileShader(const std::string& source, const std::st
...
 			pShaders->Release();
-		ThrowIfFailed(hr);
+			ThrowIfFailed(hr);
```
Comment (Skratzy) [inline Sail/src/API/DX12/shader/DX12ShaderPipeline.cpp]: Indented, is it supposed to be?

File: Sail/src/API/DX12/shader/DX12StructuredBuffer.cpp [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+	void DX12StructuredBuffer::updateData(const void* newData, unsigned int numElements, int meshIndex) {
+		assert(numElements <= MAX_ELEMENTS && "Too many elements! Increase MAX_ELEMENTS in DX12StrucutedBuffer.h");
+		assert(meshIndex <= MAX_MESHES_PER_FRAME && "Too many meshes! Increase MAX_MESHES_PER_FRAME in DX12StrucutedBuffer.h");
+
+		// This method needs to be run every frame to make sure the buffer for all framebuffers are kept updated
+		auto frameIndex = m_context->getFrameIndex();
+		auto numSwapBuffers = m_context->getNumSwapBuffers();
+
+		//if (m_resourceHeapSize[frameIndex] < m_elementByteSize * MAX_ELEMENTS) {
```
Comment (Skratzy) [inline Sail/src/API/DX12/shader/DX12StructuredBuffer.cpp]: Remove unused code

File: SPLASH/src/game/states/GameState.cpp (hunk 1 above)
Comment (Piratkopia13) [reply SPLASH/src/game/states/GameState.cpp]: H3nx will remove or use this in his branch

File: Sail/src/API/DX12/DX12API.h (hunk 2 above)
Comment (Piratkopia13) [reply Sail/src/API/DX12/DX12API.h]: yes

**Pull request merged**